File: scripts/Find All Numbers Disappeared in an Array/Find All Numbers Disappeared in an Array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> res;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            if (nums[abs(nums[i]) - 1] > 0) {
                nums[abs(nums[i]) - 1] *= -1;
            }
        }
        for (int i = 0; i < n; i++) {
            if (nums[i] > 0)
                res.push_back(i + 1);
        }
        return res;
    }
};

File: scripts/Find All Numbers Disappeared in an Array/Find All Numbers Disappeared in an Array.js
var findDisappearedNumbers = function(nums) {
    let result = [];
    for(let i = 0; i < nums.length; i++) {
        let id = Math.abs(nums[i]) - 1;
        nums[id] = - Math.abs(nums[id]);
    }
    for(let i = 0; i < nums.length; i++) {
        if(nums[i] > 0) result.push(i + 1);
    }
    return result;
};
